optimize/backend/src/main/java/io/camunda/optimize/service/importing/ImportSchedulerManagerService.java

```diff
@@ -172,6 +173,14 @@ private List<ImportMediator> createZeebeMediatorList(
         beanFactory.getBean(ZeebeImportIndexHandlerProvider.class, zeebeDataSourceDto));
 
     return zeebeMediatorFactories.stream()
+        // https://github.com/camunda/product-hub/issues/2891
+        // If the factory is not a ZeebeVariableImportMediatorFactory, object passes through
+        // If the factory is a ZeebeVariableImportMediatorFactory, it only passes when enabled
+        // When disabled, ZeebeVariableImportMediatorFactory objects are filtered out
+        .filter(
+            factory ->
+                !(factory instanceof ZeebeVariableImportMediatorFactory)
+                    || configurationService.getConfiguredZeebe().isVariablesImportEnabled())
```

Comment:
I think the logic looks reasonable to me for a quick fix. However, it might make more sense to instead handle this in the constructor, like this:
```
this.zeebeMediatorFactories = zeebeMediatorFactories.stream()
        .filter(
            factory ->
                !(factory instanceof ZeebeVariableImportMediatorFactory)
                    || configurationService.getConfiguredZeebe().isVariablesImportEnabled())
        .collect(Collectors.toList());
```

Then if anything else accesses the `zeebeMediatorFactories` variable, it will already exclude the mediators as per config. Wdyt?